doc/ssl/SSL_read.pod

```diff
@@ -2,82 +2,100 @@
 
 =head1 NAME
 
-SSL_read - read bytes from a TLS/SSL connection
+SSL_read_ex, SSL_read - read bytes from a TLS/SSL connection
 
 =head1 SYNOPSIS
 
  #include <openssl/ssl.h>
 
+ int SSL_read_ex(SSL *ssl, void *buf, size_t num, size_t *read);
  int SSL_read(SSL *ssl, void *buf, int num);
 
+ int SSL_peek_ex(SSL *ssl, void *buf, size_t num, size_t *read);
+ int SSL_peek(SSL *ssl, void *buf, int num);
+
 =head1 DESCRIPTION
 
-SSL_read() tries to read B<num> bytes from the specified B<ssl> into the
-buffer B<buf>.
+SSL_read_ex() and SSL_read() try to read B<num> bytes from the specified B<ssl>
+into the buffer B<buf>. On success SSL_read_ex() will store the number of bytes
+actually read in B<*read>.
+
+SSL_peek_ex() and SSL_peek() are identical to SSL_read_ex() and SSL_read()
+respectively except no bytes are actually removed from the underlying BIO during
+the read, so that a subsequent call to SSL_read_ex() or SSL_read() will yield
+the same bytes.
 
 =head1 NOTES
 
-If necessary, SSL_read() will negotiate a TLS/SSL session, if
+In this notes section all comments that apply to SSL_read_ex() or SSL_read()
+also apply to SSL_peek_ex() and SSL_peek().
+
+If necessary, SSL_read_ex() or SSL_read() will negotiate a TLS/SSL session, if
 not already explicitly performed by L<SSL_connect(3)> or
 L<SSL_accept(3)>. If the
 peer requests a re-negotiation, it will be performed transparently during
-the SSL_read() operation. The behaviour of SSL_read() depends on the
-underlying BIO.
+the SSL_read_ex() or SSL_read() operation. The behaviour of SSL_read_ex() and
+SSL_read() depends on the underlying BIO.
 
 For the transparent negotiation to succeed, the B<ssl> must have been
 initialized to client or server mode. This is being done by calling
 L<SSL_set_connect_state(3)> or SSL_set_accept_state()
-before the first call to an SSL_read() or L<SSL_write(3)>
-function.
+before the first call to an SSL_read_ex(), SSL_read(), L<SSL_write_ex(3)> or
+L<SSL_write(3)> function.
 
-SSL_read() works based on the SSL/TLS records. The data are received in
-records (with a maximum record size of 16kB for SSLv3/TLSv1). Only when a
-record has been completely received, it can be processed (decryption and
+SSL_read_ex() and SSL_read() work based on the SSL/TLS records. The data are
+received in records (with a maximum record size of 16kB for SSLv3/TLSv1). Only
```

Comment:
drop the "for ..." at least kill mention of SLSv3, which then makes it a truism so just drop the "for..." part.